test/plain-list/AutosuggestApp.test.js

```diff
@@ -229,6 +479,25 @@ describe('Plain list Autosuggest', () => {
     });
   });
 
+  describe('when pressing Enter and selectFirstSuggestion is true', () => {
+    before(() => {
+      setSelectFirstSuggestion(true);
+    });
+
+    after(() => {
+      setSelectFirstSuggestion(false);
+    });
+
+    beforeEach(() => {
+      focusAndSetInputValue('p');
+    });
+
+    it('should hide suggestions since the first suggestion is focused', () => {
+      clickEnter();
+      expectSuggestions([]);
+    });
```

Comment:
This test was actually added during discussion [here](https://github.com/moroshko/react-autosuggest/pull/165/commits/36bb114effa5d2ef6a4b1ed96f018138300c1440#diff-d6150ad17107b5367bf67fd9cae74c09R257). So yes, this one is actually necessary. 

What's important to note is that there's different internal logic when pressing up/down to select a suggestion and when auto-selecting or mousing over a suggestion. So it's not correct to assume that tests using keyboard inputs are valid/repetitive tests for mouse inputs or system actions. They're truly handled differently and need to be tested independently.